doc/developer-guides/coding_guidelines.rst

```diff
@@ -3141,6 +3141,98 @@ Compliant example::
        // This is a comment
 
 
+CS-18: Function information shall be documented with doxygen-style comments
+===========================================================================
+
+Some detailed rules are listed below to illustrate the comments format for each
+function:
+
+1) The comments block shall start with '/\*\*'(slash-asterisk-asterisk) in a
+   single line.
+2) The comments block shall end with ' \*/'(space-asterisk-slash) in a single
+   line.
+3) Other than the first line and the last line, every line inside the comments
+   block shall start with ' \*'(space-asterisk). It also applies to the line which
+   is used to separate different paragraphs, calling it as blank line for
+   simplicity.
+4) For each function, the information shall be documented with the following
+   order: brief description, detailed description, parameters description,
+   pre-conditions, post-conditions, return value description, and comments
+   explaining the actual return values. Calling each kind of information as
+   paragraph for simplicity. Paragraph is allowed to be removed from the list if it
+   is not applicable for specific functions.
+5) Each line shall only contain the description for one parameter, or one
+   pre-condition, or one post-condition, or one actual return value. Calling each
+   of them as element for simplicity.
+6) A blank line shall exist to separate different paragraphs. Inside each
+   paragraph, blank line is not required to separate each element.
+7) The brief description of the function shall be documented with the format
+   '@brief <brief description>'.
+8) No specific format is required for the detailed description of the function.
+9) The description of the function parameter shall be documented with the format
+   '@param <parameter name> <parameter description>'.
+10) The pre-condition of the function shall be documented with the format '@pre
+    <pre-condition description>'.
+11) The post-condition of the function shall be documented with the format
+    '@post <post-condition description>'.
+12) The brief description of the function return value shall be documented with
+    the format '@return <brief description of return value>'.
+13) A void-returning function shall be documented with the format '@return
+    None'.
+14) The comments explaining the actual return values shall be documented with
+    the format '@retval <return value> <return value explanation>'.
+15) If the description of one element needs to span multiple lines, each line
+    shall be aligned to the start of the description in the first line for that
+    element.
+
```

Comment:
Should also mention that this comment block must appear immediately before the function declaration, and in the header file.